src/main/java/com/gruzini/tennistico/services/entity_related/CourtService.java

```diff
@@ -28,27 +26,17 @@ public Court getById(Long courtId) {
     }
 
     public Court getCourt(final Court court) {
-        final Long courtId = getExistingCourtId(court);
-        if(courtId == null){
-            return courtRepository.save(court);
-        }
-        return getById(courtId);
+        final Optional<Court> existingCourt = getExistingCourt(court);
+        return existingCourt.orElseGet(() -> courtRepository.save(court));
```

Comment:
Swoją drogą bardzo dobrą lekcję dostałem na temat różnicy między Optional.orElse a Optional.orElseGet.

Reply:
wyjaśnisz trochę?

Reply:
Różnica jest taka, że orElse zawsze przygotuje swoją zawartość, nawet jeśli jej nie zwróci, bo Optional nie jest empty. Tutaj ma to gigantyczne znaczenie, bo zwracamy wynik courtRepository.save, więc obiekt, który sprawdzamy, czy już siedzi w DB (jeśli tak, to jest właśnie przez Optional zwracany) ZAWSZE znowu zapiszemy do DB.
orElseGet wykonuje swoją zawartość dopiero PO sprawdzeniu, czy Optional jest pusty.

Reply:
czyli nie zapisze znowu tylko w jedynie w przypadku jak optional okaze się pusty jest pusty, tak? Sprytne

Reply:
Znaczy nie tyle sprytne, co strasznie się głowiłem dlaczego się zapisywał w bazie nowy rekord mimo tego że identyczny już tam siedział (jak korzystałem z orElse). Ale jak widać różnica między orElse a orElseGet może być żadna albo gigantyczna. Człowiek się całe życie uczy. Co śmieszne mam wrażenie, że Kiełek o tym wspominał, ale to były czasy jak się wydawało, że po co w ogóle te Optionale.